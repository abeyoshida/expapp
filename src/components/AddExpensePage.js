import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { startAddExpense } from '../actions/expenses';

export class AddExpensePage extends React.Component {
  
  onSubmit = (expense) => {
    this.props.startAddExpense(expense);  // this will dispatch the addExpense call with the expense argument
    this.props.history.push('/dashboard');
  };
  render() {
    return (
      <div>
        <div className="page-header">
          <div className="content-container">
            <h1 className="page-header__title">Add Expense</h1>  
          </div>
        </div>
        <div className="content-container">
          <ExpenseForm
            onSubmit={this.onSubmit}
          />
        </div>
        
      </div>
    )
  }
};

const mapDispatchToProps = (dispatch) => ({
   // the mapDispatchToProps passes the dispatch call to startAddExpense
   // startAddExpense will now be available to the component via this.props.addExpense
   startAddExpense: (expense) => dispatch(startAddExpense(expense)) 
});

export default connect(undefined, mapDispatchToProps)(AddExpensePage);


