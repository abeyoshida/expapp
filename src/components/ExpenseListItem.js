import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';

const ExpenseListItem = ({ id, description, amount, createdAt }) => {
  //const createdAtSafe = (moment.isMoment(createdAt)) ? createdAt.format("dddd, MMMM Do YYYY, h:mm:ss a") : createdAt;
  const createdAtSafe = '';
  return (
      <div>
        <Link to={`/edit/${id}`}>
          <h3>{description}</h3>
        </Link>
        {/*<p>{amount} - {createdAt}</p>*/}
        <p>{amount} - {createdAtSafe}</p>
      </div>
    );
}
export default ExpenseListItem;
