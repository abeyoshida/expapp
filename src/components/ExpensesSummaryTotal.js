import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses-total';

class ExpensesSummaryTotal extends React.Component {
    constructor(props) {
        super(props);
    };
    
    render() {
        return (
            <div>
                {
                    (this.props.expensesCount === 0) ? <p>There are no expenses</p> : <p>The total amount of your expenses: {this.props.expensesTotal}</p>
                }
                
            </div>
        )
    }
};

const mapStateToProps = (state) => {
    const expensesTotal = selectExpensesTotal(state.expenses);
    return {
        expenses: selectExpenses(state.expenses, state.filters),
        expensesCount: state.expenses.length,
        expensesTotal: numeral(expensesTotal/100).format('$0,0.00')
    }
};

export default connect(mapStateToProps)(ExpensesSummaryTotal);