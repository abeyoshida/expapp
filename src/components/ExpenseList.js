import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';

// this first named export is for testing an version that is not connected to the store
export const ExpenseList = (props) => {
  //console.log('ExpenseList.js ExpenseList props: ', props);
  return (
    <div className="content-container">
      <div className="list-header">
        <div className="show-for-mobile">Expenses</div>
        <div className="show-for-desktop">Expense</div>
        <div className="show-for-desktop">Amount</div>
      </div>
      <div className="list-body">
        {
          (props.expenses.length === 0) ? (
            <div className="list-item list-item__message">
              <span>There are no expenses.</span>
            </div>
          ) : (
            props.expenses.map((expense) => {
              return <ExpenseListItem key={expense.id} {...expense} />;
            })
          )
        }
      </div>
    </div>
  );
};

const mapStateToProps = (state) => {
  //console.log('ExpenseList.js mapStateToProps state: ', state);
  return {
    expenses: selectExpenses(state.expenses, state.filters)
    // expenses: state.expenses
    
  };
};

// this export default is for the app and it uses store data
export default connect(mapStateToProps)(ExpenseList);
