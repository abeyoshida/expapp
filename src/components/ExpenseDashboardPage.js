import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import selectExpensesTotal from '../selectors/expenses-total';
import ExpensesSummaryTotal from './ExpensesSummaryTotal';


const ExpenseDashboardPage = () => (
  <div>
    <ExpensesSummaryTotal />
    <ExpenseListFilters />
    <ExpenseList />
  </div>
);

export default ExpenseDashboardPage;
