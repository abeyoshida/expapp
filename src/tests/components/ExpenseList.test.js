import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseList } from '../../components/ExpenseList';
import expensesDummy from '../fixtures/expenses';

test('render ExpenseList from expensesDummy', () => {
    const wrapper = shallow(<ExpenseList expenses={expensesDummy} />);
    expect(wrapper).toMatchSnapshot();
});
test('render ExpenseList with no expenses', () => {
    const wrapper = shallow(<ExpenseList expenses={[]} />);
    expect(wrapper).toMatchSnapshot();
});