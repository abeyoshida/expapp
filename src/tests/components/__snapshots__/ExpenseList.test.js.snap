// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render ExpenseList from expensesDummy 1`] = `
<div
  className="content-container"
>
  <div
    className="list-header"
  >
    <div
      className="show-for-mobile"
    >
      Expenses
    </div>
    <div
      className="show-for-desktop"
    >
      Expense
    </div>
    <div
      className="show-for-desktop"
    >
      Amount
    </div>
  </div>
  <div
    className="list-body"
  >
    <ExpenseListItem
      amount={225}
      createdAt={0}
      description="coffee"
      id="1"
      key="1"
      note=""
    />
    <ExpenseListItem
      amount={100000}
      createdAt={-345600000}
      description="rent"
      id="2"
      key="2"
      note=""
    />
    <ExpenseListItem
      amount={5000}
      createdAt={345600000}
      description="groceries"
      id="3"
      key="3"
      note=""
    />
  </div>
</div>
`;

exports[`render ExpenseList with no expenses 1`] = `
<div
  className="content-container"
>
  <div
    className="list-header"
  >
    <div
      className="show-for-mobile"
    >
      Expenses
    </div>
    <div
      className="show-for-desktop"
    >
      Expense
    </div>
    <div
      className="show-for-desktop"
    >
      Amount
    </div>
  </div>
  <div
    className="list-body"
  >
    <div
      className="list-item list-item__message"
    >
      <span>
        There are no expenses.
      </span>
    </div>
  </div>
</div>
`;
